baxter_collaboration_lib/include/robot_interface/robot_interface.h

```diff
@@ -116,7 +119,12 @@ class RobotInterface
     /**
      * Cartesian Controller server
      */
-    ROSThreadObj ctrl_thread;   // Internal thread that implements the controller server
+    //ROSThreadObj ctrl_thread;   // Internal thread that implements the controller server
+    std::thread ctrl_thread;
+
+    bool _thread_kill = false;
```

Comment:
remove initial underscore, and call this `ctrl_thread_close_flag`. Plus, you should not assign values to variable in the header file, but in the constructor!